### PR TITLE
CODENVY-70: add service to check if Codenvy usage is legal

### PR DESCRIPTION
**Pull request by dmytro-ndp**
@tolusha: please, review this request.

### CODE REVIEW

File: installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java [hunk 1]
```diff
@@ -81,6 +83,36 @@ public LicenseType getLicenseType() {
...
+     * 4) if (PRODUCT_KEY            IS     expired) AND (actual number of users <= MIN_NUMBER_OF_FREE_USERS)
+     */
+    public boolean isLicenseUsageLegal(long actualUsers, int actualServers) {
+        if (isExpired()) {
+            switch (getLicenseType()) {
+                case EVALUATION_PRODUCT_KEY:
+                default:
+                    return isFreeUsageLegal(actualUsers, actualServers);
+
+                case PRODUCT_KEY:
```
Comment (tolusha) [inline installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java]: Move default case to the very end of switch block.

File: installation-manager-server-war/src/test/java/com/codenvy/im/service/BaseContractTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.im.service;
+
+/**
+ * @author Dmytro Nochevnov
```
Comment (tolusha) [inline installation-manager-server-war/src/test/java/com/codenvy/im/service/BaseContractTest.java]: Duplicates

File: installation-manager-server-war/src/main/java/com/codenvy/im/service/LicenseService.java [hunk 3]
```diff
@@ -161,4 +167,41 @@ public Response getLicenseProperties() throws ApiException {
             throw new ServerException(e.getMessage(), e);
         }
     }
+
+    @GET
+    @Path("/is-codenvy-usage-legal")
```
Comment (tolusha) [inline installation-manager-server-war/src/main/java/com/codenvy/im/service/LicenseService.java]: Is it better? @Path("/legal") => GET /license/usage

File: installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java [hunk 4]
```diff
@@ -81,6 +83,36 @@ public LicenseType getLicenseType() {
...
+        }
+
+        return actualUsers <= getNumberOfUsers();
+    }
+
+    /**
+     * @return false if (actual number of users > MIN_NUMBER_OF_FREE_USERS) OR (actual number of nodes > MIN_NUMBER_OF_FREE_SERVERS)
+     */
+    public static boolean isFreeUsageLegal(long actualUsers, int actualServers) {
+        return actualUsers <= MIN_NUMBER_OF_FREE_USERS
```
Comment (sleshchenko) [inline installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java]: Why MIN_NUMBER_OF_FREE_USERS? I understand this constant as maximum number of users for free usage.

Comment (dmytro-ndp) [inline installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java]: You are right. Thank you for notice.

File: installation-manager-server-war/src/main/java/com/codenvy/im/service/LicenseService.java (hunk 3 above)
Comment (dmytro-ndp) [inline installation-manager-server-war/src/main/java/com/codenvy/im/service/LicenseService.java]: */legality* is better contraction

File: installation-manager-server-war/src/test/java/com/codenvy/im/service/BaseContractTest.java (hunk 2 above)
Comment (dmytro-ndp) [inline installation-manager-server-war/src/test/java/com/codenvy/im/service/BaseContractTest.java]: I will fix.

File: installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java (hunk 1 above)
Comment (dmytro-ndp) [inline installation-manager-core/src/main/java/com/codenvy/im/license/CodenvyLicense.java]: I will fix it.

**Pull request merged**